pkg/configv2/types/storage.go

```diff
@@ -0,0 +1,44 @@
+package types
+
+// Storage represents the configuration for various storage providers.
+// It includes settings for HTTP, IPFS, Local, and S3 providers.
+type Storage struct {
+	// HTTP is the configuration for the HTTP storage provider.
+	HTTP HTTPStorage
+	// IPFS is the configuration for the IPFS storage provider.
+	IPFS IPFSStorage `yaml:"IPFS,omitempty"`
+	// Local is the configuration for the Local storage provider.
+	Local LocalStorage `yaml:"Local,omitempty"`
+	// S3 is the configuration for the S3 storage provider.
+	S3 S3Storage `yaml:"S3,omitempty"`
+}
+
+type HTTPStorage struct {
+	// Enabled specifies whether the HTTP provider is enabled.
+	Enabled bool `yaml:"Enabled,omitempty"`
+}
+
+type IPFSStorage struct {
+	// Enabled specifies whether the IPFS provider is enabled.
+	Enabled bool `yaml:"Enabled,omitempty"`
+	// Endpoint specifies the endpoint URL for the IPFS provider.
+	Endpoint string `yaml:"Endpoint,omitempty"`
```

Comment:
if endpoint is not empty, then enabled, right?

Reply:
I am starting to like more the current behaviour where all engines are enabled by default, except if explicitly disabled, not configured (no ipfs connect) or not found (no docker installed). This is also the behaviour proposed in https://www.notion.so/expanso/Rethinking-Configuration-435fbe87419148b4bbc5119d413786eb